### PR TITLE
Log how much time spent on each label

### PR DESCRIPTION
**Pull request by enescakir**
We log the duration once the strand has finished running, but this only pertains to a single run. Currently, we lack a metric to determine the time spent on each label, which is crucial for identifying the label that consumes most of our time. This PR adds the last label change timestamp to the stack and logs the duration when the strand is hopped or exited.

### CODE REVIEW

File: model/strand.rb [hunk 1]
```diff
@@ -128,10 +133,18 @@ def unsynchronized_run
       changed_columns.delete(:schedule)
       e
     rescue Prog::Base::Hop => hp
+      last_changed_at = Time.parse(stack.first["last_label_changed_at"])
+      Clog.emit("hopped") { {strand_hopped: {duration: Time.now - last_changed_at, prog_label: prog_label}} }
```
Comment (bsatzger) [inline model/strand.rb:137]: should we also include the new label?

Comment (enescakir) [reply model/strand.rb]: Makes sense. I added it

**Review by bsatzger — APPROVED** [review]

**Review by fdr — COMMENTED** [review]
does the metric do what you want when applied recursively, e.g. after a pop?

**Review by byucesoy — APPROVED** [review]

**Comment by enescakir** [comment]
> does the metric do what you want when applied recursively, e.g. after a pop?

@fdr, currently, pushing from l1 to l2 results in three durations being printed. The first reflects the duration of the initial part of l1 when we push from l1 to l2. The second shows the duration of the l2 label when we pop from l2 back to l1. The final duration represents the second part of l1 when we move from l1 to any other label. I've considered whether we should only print two durations. While possible, this change would mean that any push operation would actually reflect the timing of the previous label, and complicating the code further. When I deploy my continuous push PR (it waited too long because of me :(), it will display two durations anyway: the duration of the l1 label before the push, and the duration of the l2 label before the pop.

**Pull request merged**